File: Labs/Lab1/TwoSum.py
def twoSum(self, nums: List[int], target: int) -> List[int]:
    sumMap = {}
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement in sumMap:
            return [sumMap[complement], i]
        sumMap[nums[i]] = i
    return []

File: Labs/Lab1/TwoSum.java
package Lab.Lab1;
import java.util.*;

public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> sum = new HashMap<>();
        for (int i = 0; i < nums.length; i++)
        {
            //value is 9-i
            int complement = target - nums[i];
            if (sum.containsKey(complement))
            {
                return new int [] {sum.get(complement),nums[i]};
            }
            sum.put(nums[i],i);
        }
        return new int[]{};
        
    }
}
